lib/server/db/postgres/postgres_test.go

```diff
@@ -127,19 +126,40 @@ var _ = Describe("Postgres", func() {
 	})
 
 	Context("creating fabric ca database", func() {
-		It("returns an error if unable execute create fabric ca database sql", func() {
-			mockDB.ExecReturns(nil, errors.New("error creating database"))
-			db.SqlxDB = mockDB
-			_, err := db.CreateDatabase()
-			Expect(err).To(HaveOccurred())
-			Expect(err.Error()).Should(ContainSubstring("Failed to create Postgres database: Failed to execute create database query: error creating database"))
-		})
-
-		It("creates the fabric ca database", func() {
-			db.SqlxDB = mockDB
-
-			_, err := db.CreateDatabase()
-			Expect(err).NotTo(HaveOccurred())
+		When("creating the Fabric CA database fails", func() {
+			It("returns an error", func() {
+				mockDB.ExecReturns(nil, errors.New("error creating database"))
+				db.SqlxDB = mockDB
+				_, err := db.CreateDatabase()
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(Equal("Failed to create Postgres database: Failed to execute create database query: error creating database"))
+			})
+		})
+
+		When("the database does not exist", func() {
+			It("creates the fabric ca database", func() {
+				db.SqlxDB = mockDB
+				sqlxDB, err := db.CreateDatabase()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(db.SqlxDB).To(Equal(sqlxDB))
+				Expect(mockDB.ExecCallCount()).To(Equal(1))
+			})
+		})
+
+		When("the database already exists", func() {
+			It("does not attempt to create the database", func() {
+				mockDB.GetCalls(func(s string, i interface{}, s2 string, i2 ...interface{}) error {
+					exists := i.(*bool)
+					*exists = true
+					i = exists
+					return nil
```

Comment:
Not sure this is the best way to do it, but this is what I came up with to make sure the value assigned to the interface sticks.